presto-spi/src/main/java/io/prestosql/spi/predicate/TupleDomain.java

```diff
@@ -76,17 +80,23 @@ private TupleDomain(Optional<Map<T, Domain>> domains)
 
     public static <T> TupleDomain<T> withColumnDomains(Map<T, Domain> domains)
     {
-        return new TupleDomain<>(Optional.of(requireNonNull(domains, "domains is null")));
+        requireNonNull(domains, "domains is null");
+        if (domains.isEmpty()) {
```

Comment:
what about none?

Reply:
This method takes `Map`, not `Optinal(Map)` so there is no `none` case here.